python/py_vapid/tests/test_vapid.py

```diff
@@ -247,3 +252,29 @@ def test_invalid_sig(self, mm):
                           decode,
                           'foo.bar.a',
                           'aaaa')
+
+    def test_sub(self):
```

Comment:
Rarely do I use py.test's parameterized tests (it's a bit of magic) but I may have actually considered it here

Reply:
Maybe? I dunno. For something like this, I think it's clearer for future me to be less clever and just specify what goes into which gauntlet as a delineated list like this, rather than as a CSV.

Reply:
Yea that's part of why I don't use it, it's a bit too clever. I think the main benefit is debugging and test output is a bit improved over a loop